src/lib/authentication/msalAuthProvider.ts

```diff
@@ -208,22 +171,44 @@ export class MsalAuthProvider implements AuthProvider {
     }
     return false;
   }
-  getLoginStatus(): LoginStatus {
-    if (this.myMSALObj) {
-      const isLoggedIn = this.myMSALObj.getAllAccounts().length > 0;
-      this.loginStatus = isLoggedIn
-        ? LoginStatus.Logged
-        : LoginStatus.NotLogged;
-      this.notifySubscribers();
 
-      return this.loginStatus;
+  //PRIVATE METHODS
+
+  private handleResponse(response: msal.AuthenticationResult | null) {
+    if (response !== null) {
+      this.account = response.account;
+    } else {
+      const accounts = this.myMSALObj!.getAllAccounts();
+      this.account = accounts ? accounts[0] : null;
     }
-    return LoginStatus.NotLogged;
   }
-  onLoginStatus(subscriber: (status: string) => void) {
-    this.subscribers.add(subscriber);
-    return () => {
-      this.subscribers.delete(subscriber);
-    };
+  private async getProfileTokenRedirect(): Promise<string | null> {
+    if (this.account) {
+      this.silentProfileRequest.account = this.account;
+    }
+    return this.getTokenRedirect(
+      this.silentProfileRequest,
+      this.profileRedirectRequest
+    );
+  }
+
+  private async getTokenRedirect(
+    silentRequest: msal.SilentRequest,
+    interactiveRequest: msal.RedirectRequest
+  ): Promise<string | null> {
+    try {
+      const response = await this.myMSALObj!.acquireTokenSilent(silentRequest);
+
+      return response.accessToken;
+    } catch (e) {
+      if (e instanceof msal.InteractionRequiredAuthError) {
+        this.myMSALObj!.acquireTokenRedirect(interactiveRequest).catch((e) => {
```

Comment:
I'm not convinced. does this means that we trigger a 'login()' flow with redirects, which include the 'unload' off the App and getting back to the 'returnUri' in case the accessToken expired during a request to the backend?

I'm not convinced is the right approach to unload the application and reload potentially during a background activity like 'polling' to keep a table contents fresh.

if you tested and behaviours are sound, for me is ok. please make sure to test this behaviour.

Reply:
This is straight out of the documentation example of msal-browser. 
`"Gets the token to read user profile data from MS Graph silently, or falls back to interactive redirect."`